openid-4-verifiable-presentations-1_0.md

```diff
@@ -559,11 +565,276 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+supported Credential Formats as defined in the Wallet's metadata.
+
+`expect_meta`: 
+: OPTIONAL. An object defining additional properties requested by the Verifier that
+apply to the metadata and validity data of the Credential. The properties of
+this object are defined per Credential Format in (#format_specific_properties). If omitted,
+no specific constraints are placed on the metadata or validity of the requested
+Credential.
+
+`expect_claims`:
+: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
+claims in the requested Credentials.
+
+`valid_claim_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for elements in
+`expect_claims`. The identifier MAY be postfixed by `?`, indicating that
+delivery of the the respective claim is OPTIONAL.  
+
+## Claims Query {#claims_query}
+
+Each entry in `expect_claims` MUST be an object with the following properties:
+
+`id`:
+: REQUIRED if `valid_claim_sets` is present; OPTIONAL otherwise. A string
+identifying the particular claim. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the particular `expected_claims` array, the same `id` MUST NOT
+be present more than once.
+
+`path`:
+: REQUIRED if the Credential Format uses a JSON-based claims structure; MUST NOT
+be present otherwise. The value MUST be a claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+`namespace`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the claim
+within the Verifiable Credential, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the name of the claim within the provided namespace
+in the Verifiable Credential, e.g., `first_name`.
+
+`value`:
+: OPTIONAL. A string, integer or boolean value that specifies the expected value of the claim. If the
+`value` property is present, the Wallet MUST return the claim only if the type and value
+of the claim match the type and value specified in the query.
+
+`values`:
+: OPTIONAL. An array of strings, integers or boolean values that specifies the expected values of the claim.
+If the `values` property is present, the Wallet MUST return the claim only if the
+type and value of the claim both match for at least one of the elements in the array.
+
+### Selecting Claims and Credentials {#vp_query_lang_processing_rules}
+
+The same basic logic applies for selecting claims and for selecting credentials, as detailed in the following.
+
+#### Selecting Claims
+
+The following rules apply for selecting claims via `expect_claims` and `valid_claim_sets`:
```

Comment:
Like `expect_` wonder if we should just have `claim_sets` and `credential_sets` instead.

Reply:
Agree, I dropped the `valid_` prefix.